pkg/node/node.go

```diff
@@ -25,9 +25,43 @@ type Node struct {
 
 // Lookup queries nodes looking for a particular node in the network
 func (n *Node) Lookup(ctx context.Context, to pb.Node, find pb.Node) ([]*pb.Node, error) {
+	c, err := n.getConnection(ctx, to)
```

Comment:
nothing is closing the connection or putting it back into the cache

Reply:
getConnection adds it to the cache if the connection succeeds

Reply:
Ah, so a concurrent use of `Node` isn't allowed? I'm just thinking whether this the assumption that the rest of the system is having.

Let's say start storj as a library and start multiple concurrent uploads, so would it cause problems here?

Reply:
Because multiple goroutines calling Lookup would end up with the same connection. Ah, wait, does grpc handle it transparently already?

Reply:
But, there's still a possibility for a connection leak when multiple goroutines create a new connection, then only one of them get added to the cache.

Reply:
the cache has a read/write mutex which would prevent that thought right?

Reply:
no, because connecting to the node is not done inside the mutex

Reply:
ahh very good catch -- you are correct. Let me fix that.